Tests/tests/test_MotionDetection.cpp

```diff
@@ -96,107 +95,212 @@ TEST_F(MotionDetectionEventTest, getMotionDetectors)
+                return MOTION_DETECTION_RESULT_SUCCESS;
+            }));
 
-        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("arm"), _T("{\"index\":\"FP_MD\",\"mode\":\"10\" }"), response));
-        EXPECT_EQ(response,  string("{\"success\":false}"));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("disarm"), _T("{\"index\":\"FP_MD\"}"), response));
+    EXPECT_EQ(response,  string("{\"success\":true}"));
 }
 
-
-TEST_F(MotionDetectionEventTest, disarm)
+TEST_F(MotionDetectionEventTest, disarmInvalid)
 {
+    ON_CALL(motionDetectionImplMock, MOTION_DETECTION_DisarmMotionDetector(::testing::_))
+        .WillByDefault(::testing::Invoke(
+            [](std::string index) {
+                return MOTION_DETECTION_RESULT_INDEX_ERROR;
+            }));
+
+
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("disarm"), _T("{\"index\":\"FP_XX\"}"), response));
+    EXPECT_EQ(response,  string(""));
 
-        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("disarm"), _T("{\"index\":\"FP_MD\"}"), response));
-        EXPECT_EQ(response,  string("{\"success\":true}"));
+    ON_CALL(motionDetectionImplMock, MOTION_DETECTION_DisarmMotionDetector(::testing::_))
+        .WillByDefault(::testing::Invoke(
+            [](std::string index) {
+                return MOTION_DETECTION_RESULT_INDEX_ERROR;
+            }));
+
+
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("disarm"), _T("{}"), response));
+    EXPECT_EQ(response,  string(""));
 }
-TEST_F(MotionDetectionEventTest, disarm)
+
+TEST_F(MotionDetectionEventTest, isarmed)
 {
+    ON_CALL(motionDetectionImplMock, MOTION_DETECTION_IsMotionDetectorArmed(::testing::_,::testing::_))
+        .WillByDefault(::testing::Invoke(
+            [](std::string index, bool *isArmed) {
+                *isArmed = false;
+                return MOTION_DETECTION_RESULT_SUCCESS;
+            }));
+
 
-        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("disarm"), _T("{\"index\":\"FP_MD\"}"), response));
-        EXPECT_EQ(response,  string("{\"success\":true}"));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("isarmed"), _T("{\"index\"=\"FP_MD\"}"), response));
+    EXPECT_EQ(response,  string("{\"state\":false,\"success\":true}"));
 }
 
-TEST_F(MotionDetectionEventTest, disarmError)
+TEST_F(MotionDetectionEventTest, isarmedInvalid)
 {
+    ON_CALL(motionDetectionImplMock, MOTION_DETECTION_IsMotionDetectorArmed(::testing::_,::testing::_))
+        .WillByDefault(::testing::Invoke(
+            [](std::string index, bool *isArmed) {
+                return MOTION_DETECTION_RESULT_INDEX_ERROR;
+            }));
 
-        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("disarm"), _T("{\"index\":\"FP_XX\"}"), response));
-        EXPECT_EQ(response,  string("{\"success\":false}"));
-}
 
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("isarmed"), _T("{}"), response));
```

Comment:
is it returning ERROR_GENERAL due to invalid parameters or due to MOTION_DETECTION_RESULT_INDEX_ERROR? This is not clear from the test or its name